File: 1044-find-common-characters/1044-find-common-characters.py
class Solution(object):
    def commonChars(self, words):
        seen = [float('inf')] * 26
        for s in words:
            count = [0] * 26
            for c in s:
                count[ord(c) - ord('a')] += 1
            for i in range(26):
                seen[i] = min(seen[i], count[i])
        
        ans = []
        for i in range(26):
            for j in range(seen[i]):
                ans.append(chr(ord('a') + i))
        
        return ans
        

File: 1044-find-common-characters/1044-find-common-characters.cpp
class Solution {
public:
    vector<string> commonChars(vector<string>& words) {
        vector<int> seen(26, INT_MAX);
        for(string s : words) {
            vector<int> count(26, 0);
            for(char c : s) {
                count[c - 'a']++;
            }
            for(int i = 0; i<26; i++) {
                seen[i] = min(seen[i], count[i]);
            }
        }

        vector<string> ans;
        for(int i = 0; i<26; i++) {
            for(int j = 0; j<seen[i]; j++) {
                ans.push_back(string(1, 'a' + i));
            }
        }

        return ans;
    }
};